### PR TITLE
kernel-install: respect $MACHINE_ID, ignore /etc/machine-id if on tmpfs

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Draft because on top of #21623. See commit message for reasoning.

Extracted from #19006

### CODE REVIEW

**Comment by nabijaczleweli** [comment]
Rebased, should be g2g

File: man/kernel-install.xml [hunk 1]
```diff
@@ -161,8 +161,12 @@
 
   <refsect1>
     <title>Environment variables</title>
+
     <para>If <option>--verbose</option> is used, <varname>$KERNEL_INSTALL_VERBOSE=1</varname> will be set for
     the plugins. They may output additional logs in this case.</para>
+
+    <para>If <varname>MACHINE_ID=</varname> is set and not empty, it will be used as <replaceable>MACHINE-ID</replaceable>,
+    overriding any automatic detection attempts, and must be 32 hexadecimal characters.</para>
```
Comment (keszybz) [inline man/kernel-install.xml]: ", and must be 32 hexadecimal charcters" → ". The value must be a valid machine ID (32 hexadecimal characters)" or something like that.

File: man/kernel-install.xml [hunk 2]
```diff
@@ -214,9 +218,9 @@
         </term>
           <listitem>
             <para>The content of this file specifies the machine identification
-            <replaceable>MACHINE-ID</replaceable>.  If <filename>$BOOT/Default</filename> exists,
-            or <filename>/etc/machine-id</filename> doesn't, <command>kernel-install</command>
-            will use the literal <literal>Default</literal> as the machine ID instead.</para>
+            <replaceable>MACHINE-ID</replaceable>. If <filename>/etc/machine-id</filename>
+            cannot be read or is temporary (backed by a file on <constant>tmpfs</constant>),
+            <command>kernel-install</command> will use the literal <literal>Default</literal> instead.</para>
```
Comment (keszybz) [inline man/kernel-install.xml]: Strike "the literal" ?

**Review by keszybz — COMMENTED** [review]
Could you paste the commit justification into the commit message please?

**Comment by nabijaczleweli** [comment]
All done

**Comment by keszybz** [comment]
centos ci (arch linux):
```
01:17:32     * 
01:17:32 Failed tests: 1
01:17:32     * testcase_iscsi_lvm
01:17:32 make: *** [Makefile:4: run] Error 1
01:17:32 make: Leaving directory '/build/test/TEST-64-UDEV-STORAGE'
```
I haven't seen this particular test fail yet.

focal-ppc64el: the same hardware issues as before

**Pull request merged**